sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -354,47 +419,47 @@ export namespace ConnectionContext {
         await Promise.all(detachCalls);
       }
 
+      // Calling onDetached on batching receivers for the same reasons as sender
+      const numBatchingReceivers = getNumberOfReceivers(connectionContext, "batching");
+      if (!state.wasConnectionCloseCalled && numBatchingReceivers) {
+        logger.verbose(
+          `[${connectionContext.connection.id}] connection.close() was not called from the sdk and there were ${numBatchingReceivers} ` +
+            `batching receivers. We should reconnect.`
+        );
+
+        // Call onDetached() on receivers so that batching receivers it can gracefully close any ongoing batch operation
+        await callOnDetachedOnReceivers(
+          connectionContext,
+          connectionError || contextError,
+          "batching"
+        );
+
+        // TODO:
+        //  `callOnDetachedOnReceivers` handles "connectionContext.messageReceivers".
+        //  ...What to do for sessions (connectionContext.messageSessions) ??
```

Comment:
What to do for sessions?
(Beyond the scope of this PR and was unhandled previously too.)

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/8875#issuecomment-767210226